src/compiler/checker.ts

```diff
@@ -7269,10 +7310,17 @@ namespace ts {
             let needToCaptureLexicalThis = false;
 
             if (container.kind === SyntaxKind.Constructor) {
-                const baseTypeNode = getClassExtendsHeritageClauseElement(<ClassLikeDeclaration>container.parent);
-                if (baseTypeNode && !(getNodeCheckFlags(container) & NodeCheckFlags.HasSeenSuperCall)) {
-                    // In ES6, super inside constructor of class-declaration has to precede "this" accessing
-                    error(node, Diagnostics.super_must_be_called_before_accessing_this_in_the_constructor_of_a_derived_class);
+                const containingClassDecl = <ClassDeclaration>container.parent;
+                const baseTypeNode = getClassExtendsHeritageClauseElement(containingClassDecl);
+
+                // If a containing class does not have extends clause or the class extends null
+                // skip checking whether super statement is called before "this" accessing.
+                if (baseTypeNode && !isClassDeclarationExtendNull(containingClassDecl)) {
+                    const superStatement = getSuperStatementInConstructor(<ConstructorDeclaration>container);
+                    if (!superStatement || (superStatement && superStatement.pos > node.pos || superStatement.end > node.pos)) {

```

Comment:
You probably meant to write 

```ts
superStatement && (superStatement.pos > node.pos || superStatement.end > node.pos)
```

since `&&` binds more tightly than `||`